### PR TITLE
add test for submit component with set unknown input file.

### PR DESCRIPTION
**Pull request by Stephen1993**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py [hunk 1]
```diff
@@ -3261,3 +3275,18 @@ def pipeline_func():
             node_compute = rest_obj.properties.jobs[node_name]["computeId"]
             assert is_singularity_id_for_resource(node_compute)
             assert node_compute.endswith(singularity_vc.name)
+
+    def test_assign_value_to_unknow_filed(self, client: MLClient):
+        path = "./tests/test_configs/components/helloworld_component.yml"
+        component_func = load_component(source=path)
+        component_func.tags = {}
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py:3282]: maybe add a comment here to explain why we use this workaround?

**Review by elliotzh — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py [hunk 2]
```diff
@@ -3261,3 +3275,18 @@ def pipeline_func():
             node_compute = rest_obj.properties.jobs[node_name]["computeId"]
             assert is_singularity_id_for_resource(node_compute)
             assert node_compute.endswith(singularity_vc.name)
+
+    def test_assign_value_to_unknow_filed(self, client: MLClient):
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py:3279]: typo: unknow -> unknown

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py (hunk 1 above)
Comment (Stephen1993) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py:3283]: done

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py (hunk 2 above)
Comment (Stephen1993) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py]: done

**Pull request merged**